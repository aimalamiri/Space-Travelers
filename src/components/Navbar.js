import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navbar.css';
import { Navbar as BNavbar, Container, Nav } from 'react-bootstrap';

const planet = require('../images/planet.png');

const Navbar = () => {
  const activeLink = (isActive) => (isActive ? { borderBottom: '1px solid #444' } : undefined);

  return (
    <>
      <BNavbar collapseOnSelect expand="lg" bg="light" variant="light">
        <Container>
          <BNavbar.Brand>
            <img className="logo" src={planet} alt="space travelers" />
            <span className="mx-2">Space Travelers</span>
          </BNavbar.Brand>
          <BNavbar.Toggle aria-controls="responsive-navbar-nav" />
          <BNavbar.Collapse id="responsive-navbar-nav">
            <Nav className="ms-auto gap-4">
              <NavLink to="/" style={({ isActive }) => activeLink(isActive)} className="text-dark">
                Rockets
              </NavLink>
              <NavLink to="/missions" style={({ isActive }) => activeLink(isActive)} className="text-dark">
                Missions
              </NavLink>
              <NavLink to="/profile" style={({ isActive }) => activeLink(isActive)} className="text-dark">
                My Profile
              </NavLink>
            </Nav>
          </BNavbar.Collapse>
        </Container>
      </BNavbar>
    </>
  );
};

export default Navbar;
