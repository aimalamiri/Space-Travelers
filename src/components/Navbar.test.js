import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from './Navbar';

it('Renders Navbar', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
