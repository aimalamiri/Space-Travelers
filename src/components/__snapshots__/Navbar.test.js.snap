// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Navbar 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-light bg-light"
>
  <div
    className="container"
  >
    <span
      className="navbar-brand"
    >
      <img
        alt="space travelers"
        className="logo"
        src="planet.png"
      />
      <span
        className="mx-2"
      >
        Space Travelers
      </span>
    </span>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="responsive-navbar-nav"
    >
      <div
        className="ms-auto gap-4 navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          aria-current="page"
          className="text-dark active"
          href="/"
          onClick={[Function]}
          style={
            Object {
              "borderBottom": "1px solid #444",
            }
          }
        >
          Rockets
        </a>
        <a
          className="text-dark"
          href="/missions"
          onClick={[Function]}
        >
          Missions
        </a>
        <a
          className="text-dark"
          href="/profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </div>
    </div>
  </div>
</nav>
`;
